### PR TITLE
merge alica repos

### PR DESCRIPTION
**Pull request by bjoernschroeder**

### CODE REVIEW

File: alica-supplementary/alica_viewer/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,103 @@
+cmake_minimum_required(VERSION 3.5.1)
```
Comment (abhi1287) [inline alica-supplementary/alica_viewer/CMakeLists.txt]: kill the alica_viewer

File: alica-supplementary/travis_scripts/before_script.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
```
Comment (abhi1287) [inline alica-supplementary/travis_scripts/before_script.sh]: don't need this, main repo script would be used

File: alica-supplementary/alica_tracing/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,62 @@
+cmake_minimum_required(VERSION 2.8.3)
```
Comment (abhi1287) [inline alica-supplementary/alica_tracing/CMakeLists.txt]: this package should be called alica_jaeger_tracing

Comment (abhi1287) [reply alica-supplementary/alica_tracing/CMakeLists.txt]: can happen in a different pr though - @veerajsk

File: alica-supplementary/.clang-format [hunk 4]
```diff
@@ -0,0 +1,91 @@
+---
+Language:        Cpp
```
Comment (abhi1287) [inline alica-supplementary/.clang-format]: can kill repo level clang, gitignore, travis files

File: alica-essentials/system_util/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.5.1)
```
Comment (abhi1287) [inline alica-essentials/system_util/CMakeLists.txt]: can kill system_util

File: alica-essentials/nonstd/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,35 @@
+cmake_minimum_required(VERSION 3.5.1)
+project(nonstd)
```
Comment (abhi1287) [inline alica-essentials/nonstd/CMakeLists.txt]: can kill nonstd

File: alica-essentials/event_handling/include/essentials/NotifyTimer.hpp [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+        std::lock_guard<std::mutex> lockGuard(_cv_mtx);
+        _started = false;
+        _running = false;
+    }
+    _cv.notify_all();
+    _runThread->join();
+}
+
+template <class NotificationClass>
+bool NotifyTimer<NotificationClass>::start()
```
Comment (abhi1287) [inline alica-essentials/event_handling/include/essentials/NotifyTimer.hpp]: Use AlicaTimer in VariableSyncModule and kill dependency on NotifyTimer

File: .travis.yml [hunk 8]
```diff
@@ -68,7 +68,7 @@ script:
   - docker exec -t alica-test bash -c "
     source /opt/ros/$ROS_DISTRO/setup.bash &&
     cd ~/catkin_ws &&
-    ./src/alica-supplementary/alica_tracing/scripts/install.sh &&
+    ./src/alica/alica-supplementary/alica_tracing/scripts/install.sh &&
```
Comment (abhi1287) [inline .travis.yml]: Do we really need this? Alica and supplementary tests can supply dummy tracing. we don't need jaeger tracing in tests.

File: alica-essentials/event_handling/include/essentials/ITrigger.hpp [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+namespace essentials
+{
+
+class ITrigger
+{
+public:
+    virtual ~ITrigger() = default;
+    virtual void run(bool notifyAllThreads) = 0;
+
+    virtual void registerCV(std::condition_variable* condVar)
```
Comment (abhi1287) [inline alica-essentials/event_handling/include/essentials/ITrigger.hpp]: can kill ITrigger now

File: .travis.yml (hunk 8 above)
Comment (veerajsk) [reply .travis.yml]: this is fixed in https://github.com/rapyuta-robotics/alica/pull/184

File: alica-essentials/LICENSE [hunk 10]
```diff
@@ -0,0 +1,21 @@
+T License (MIT)
```
Comment (abhi1287) [inline alica-essentials/LICENSE]: can kill this. alica already has license which applies to full repo other than the license in alica_designer_runtime

File: alica-essentials/format_cpp.sh [hunk 11]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
```
Comment (abhi1287) [inline alica-essentials/format_cpp.sh]: can kill

File: alica-essentials/pre-commit [hunk 12]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```
Comment (abhi1287) [inline alica-essentials/pre-commit]: can kill

File: alica-supplementary/LICENSE [hunk 13]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (abhi1287) [inline alica-supplementary/LICENSE]: kill

File: alica-supplementary/alica_ros_turtlesim/CMakeLists.txt [hunk 14]
```diff
@@ -0,0 +1,67 @@
+cmake_minimum_required(VERSION 2.8.3)
```
Comment (abhi1287) [inline alica-supplementary/alica_ros_turtlesim/CMakeLists.txt]: please update main repo Readme links to point to the turtlesim demo and alica_designer_runtime folders in this repo.

**Review by veerajsk — APPROVED** [review]

**Pull request merged**